chemprop/v2/data/utils.py

```diff
@@ -2,65 +2,253 @@
 from typing import Sequence
 
 import numpy as np
-from rdkit.Chem.Scaffolds import MurckoScaffold
 
 
 from chemprop.v2.data.datapoints import MoleculeDatapoint
+from chemprop.v2.data.datasets import MoleculeDataset
+
+from astartes import train_val_test_split, train_test_split
+from astartes.molecules import train_val_test_split_molecules, train_test_split_molecules
+
+import copy
+from logging import Logger
+import pickle
+from random import Random
+from typing import Tuple
+import os
+
+from chemprop.data import log_scaffold_stats
+from chemprop.args import TrainArgs
 
 
 def split_data(
     data: Sequence[MoleculeDatapoint],
     split: str = "random",
-    sizes: tuple[float, float, float] = (0.8, 0.1, 0.1),
-    k: int = 5,
-    fold: int = 0,
-):
+    sizes: Tuple[float, float, float] = (0.8, 0.1, 0.1),
+    key_molecule_index: int = 0,
+    seed: int = 0,
```

Comment:
If the seed defaults to 0, will this give the same "random" split each time Chemprop is run? It could be better to default seed to 'None'.

Reply:
The seed should be set to 0 by default to follow chemprop's existing practice of being reproducible without needing to explicitly set a seed. People who want different splits each time will have to specify different seeds / use cross-validation.

Reply:
This is also practice in `astartes`.